File: app/src/main/java/com/sofascoreacademy/sofascoreacademy/PageViewModel.java
package com.sofascoreacademy.sofascoreacademy;

import androidx.arch.core.util.Function;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

import com.sofascoreacademy.model.Event;

public class PageViewModel extends ViewModel {

    private MutableLiveData<Integer> mIndex = new MutableLiveData<>();
    private Event[] events;
    private LiveData<String> mText = Transformations.map(mIndex, new Function<Integer, String>() {
        @Override
        public String apply(Integer input) {
            return events[input - 1].getHomeTeam().getName() + " - " + events[input - 1].getAwayTeam().getName();
        }
    });

    public void setIndexAndEvents(int index, Event[] events) {
        mIndex.setValue(index);
        this.events = events;
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/com/sofascoreacademy/sofascoreacademy/PageViewModel.kt
package com.sofascoreacademy.sofascoreacademy

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.Transformations
import androidx.lifecycle.ViewModel
import com.sofascoreacademy.model.Event

class PageViewModel : ViewModel() {
    private val mIndex = MutableLiveData<Int>()
    private lateinit var events: Array<Event>
    val text = Transformations.map(mIndex) { input -> events[input - 1].homeTeam.name + " - " + events[input - 1].awayTeam.name }

    fun setIndexAndEvents(index: Int, events: Array<Event>) {
        mIndex.value = index
        this.events = events
    }

}